src/main/kotlin/net/ccbluex/liquidbounce/features/module/modules/player/invcleaner/ModuleInventoryCleaner.kt

```diff
@@ -43,6 +49,48 @@ object ModuleInventoryCleaner : Module("InventoryCleaner", Category.PLAYER) {
 
     private val isGreedy by boolean("Greedy", true)
 
+    private var itemLimits: Map<Item, Int> = mapOf()
+    private val presentSettings: MutableList<Pair<Value<MutableList<Item>>, Value<Int>>> = mutableListOf()
+
+    private fun recount() {
+        val limits = mutableMapOf<Item, Int>()
+        presentSettings.forEach { (itemsValue, countValue) ->
+            val count = countValue.get()
+            itemsValue.get().forEach { item ->
+                val limitState = limits[item]
+                // we just follow the lowest filter
+                limits[item] = if (limitState == null) {
+                    count
+                } else {
+                    min(count, limitState)
+                }
+            }
+        }
+        itemLimits = limits
+    }
```

Comment:
> The user configures i.e.:
> 
> ```yaml
> - Firery Items:
>     - items:
>         - minecraft:flint_and_steel
>         - minecraft:lava_bucket
>     - limit: 2
> ```
> 
> As a user I'd think that the group `Firery Items` acts as a filter group and accepts 2 of _any_ of the given items. (i.e. `Lava, Lighter`) or (`Lighter, Lighter`). But the client would accept 2 of each. (i.e. `Lava, Lava, Lighter, Lighter`). Is that the desired behaviour?

I think it should be like this, may be it would be good to add ability to add new limits per category (and add various categories like Firery items (we can also possibly add ability to add new categories, but that seems kinda unnecessary))